modules/scripts/startup-script/files/setup-raid.yml

```diff
@@ -0,0 +1,96 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+---
+
+- name: Configure local SSDs
+  become: true
+  hosts: localhost
+  vars:
+    DST_MNT: "/mnt/localssd"
+    bashrc: "{{ '/etc/bashrc' if ansible_facts['os_family'] == 'RedHat' else '/etc/bash.bashrc' }}"
+    array_dev: "/dev/md0"
+    mode: 755
+  tasks:
+  - name: Check if DST_MNT exists
+    ansible.builtin.stat:
+      path: "{{ DST_MNT }}"
+    register: dst_mnt_exists
+
+  - name: Check if {{ array_dev }} exists
+    ansible.builtin.stat:
+      path: "{{ array_dev }}"
+    register: array_dev_exists
+
+  - name: Install mdadm
+    ansible.builtin.package:
+      name: mdadm
+      state: present
+
+  - name: Get number of devices
+    ansible.builtin.shell: nvme list | { grep nvme_ || test $? = 1; }  | { grep -v nvme_card-pd || test $? = 1; } | awk '{print $1}' | wc -l
+    register: nvme_device_count
+
+  - name: Get list of devices
+    ansible.builtin.shell: nvme list | { grep nvme_ || test $? = 1; }  | { grep -v nvme_card-pd || test $? = 1; } | awk '{print $1}' | paste -sd ' '
+    register: nvme_device_list
+
+  - name: Exit if less than 2 local ssd
+    ansible.builtin.meta: end_play
+    when: nvme_device_count.stdout < "2"
```

Comment:
How will the user experience differ for 1 ssd vs 2. Where will they access the the ssds in the file system. My thinking is that the user experience should not differ. If the user has to write data to a different location based on the number of local ssds, this will cause confusion.

Maybe it doesn't make sense to raid a single ssd but could we at least mount it to the same location.

Reply:
Quite a number of the `g2` VM family only support 1 local SSD disk. And in other VM types, you can configure the number of local SSD (scratch) disks to add. So 1 is a use case we should consider at least for the purpose of getting the input variable definition right.

There are a couple options here:

1. In fact you can software RAID a single SSD. `mdadm` complains at you and you must supply a `--force` (or similar) argument, but it will work. I believe it can make sense to do this in on-premises systems because it opens up the possibility of adding more disks to an existing RAID. It makes a little less sense in the cloud.
2. Alter the input variable format (see comment above) to be less focused on RAID false or true and more focused on RAID or "just a disk". I don't yet have a proposal but maybe we can find the right answer.

Reply:
Currently I have a task that is a basically a duplicate of the regular raid task, but uses `--force` since it only runs if there is 1 local ssd. This aligns with option 1 you listed above. 
```
  - name: Force RAID array if only 1 local SSD
    ansible.builtin.shell: mdadm --create "{{ array_dev }}" --name="{{ raid_name }}" --homehost=any --level=0 --raid-devices=1 /dev/disk/by-id/google-local-nvme-ssd-* --force
    args:
      creates: "{{ array_dev }}"
    when: local_ssd_devices.files | length == 1
```

Is there an alternative option that you can think of using the input variable format?